auth.go

```diff
@@ -109,12 +109,13 @@ func (b *mesosBackend) authRenew(ctx context.Context, req *logical.Request, d *f
 		return nil, err
 	}
 
-	// :-( :-( :-( :-( :-( :-( :-( :-( :-( :-( :-(
-	taskIDorNil := req.Auth.InternalData["task-id"]
-	if taskIDorNil == nil || taskIDorNil.(string) == "" {
+	// We have to check for nil before casting to avoid a segfault, and we
```

Comment:
s/and/but/?

Reply:
The two things are pretty much independent, so I think "and" is marginally better than "but" here.